### PR TITLE
Preserve the original question name in a "save question" modal

### LINKED ISSUE #41196: Suggest the original question name for derived question in a "save question" modal
[Product doc](https://www.notion.so/metabase/Suggest-original-question-name-for-derived-question-528b990f16204d91945ca2d7bb5a9ba3)

When you modify a saved question and want to save it as a new question we auto-generate a new name for it by default.
The task here is to stop auto-generating names, and to suggest the original question name instead.

Motivation:
Most of the time people just want to slightly modify the original (previous) question name.

### PR DESCRIPTION
**Pull request by nemanjaglumac**
Resolves #41196
Reproduces #41196 

### Description

We're now initializing form values with the name and the description derived from the original question. If, for whatever reason, that question name is not available, we're falling back to auto-generated descriptive question name. And if even that doesn't work, we will leave the `name` field empty for user to populate.

### How to verify

1. Open any saved question (let's say that its name was "Foo")
2. Modify anything about that question (filter, sort, row limit)
3. Click on "Save"
4. Choose "Save as new question"
5. Make sure that the suggested pre-filled name in the form is still "Foo"

Run an associated E2E reproduction.

### Demo
Kapture 2024-04-09 at 18 40 00

### Checklist
- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by Somtom — APPROVED** [review]
Nice 👍

File: frontend/src/metabase/containers/SaveQuestionModal/SaveQuestionModal.tsx [hunk 1]
```diff
@@ -113,10 +113,21 @@ export const SaveQuestionModal = ({
     }
   }
 
+  const getOriginalNameModification = (originalQuestion: Question | null) =>
+    originalQuestion
+      ? originalQuestion?.displayName() + " - " + t`Modified`
```
Comment (kamilmielnik) [inline frontend/src/metabase/containers/SaveQuestionModal/SaveQuestionModal.tsx:118]: This won't work nicely for RTL languages.
We need to make the question name a part of the translation string, something like
```tsx
t`${questionName} - Modified`
```

Comment (kamilmielnik) [inline frontend/src/metabase/containers/SaveQuestionModal/SaveQuestionModal.tsx:118]: Redundant `?`

```suggestion
      ? originalQuestion.displayName() + " - " + t`Modified`
```

Comment (nemanjaglumac) [reply frontend/src/metabase/containers/SaveQuestionModal/SaveQuestionModal.tsx:118]: Yep.
Thanks for catching. Leftover from before.

**Review by mngr — APPROVED** [review]

**Review by kamilmielnik — APPROVED** [review]
:+1:

**Pull request merged**